### DISCUSSION TITLE
TLS examples?

### DISCUSSION THREAD

**Discussion by tlewisflood** [fengtan/ldap-explorer#1] [Q&A] [answered]
I've got v1.1.1 on Windows10 in a typical corporate ActiveDirectory setup. Native AD queries work fine, but ldap-explorer doesn't pass the connection test with "Connection failed: Unable to bind: unable to get local issuer certificate". All the necessary root CAs are in windows Certificates - Local Computer\Trusted Root Certification Authorities\Certificates, but how to use them, or at least point to a cert bundle file like whatever.pem?

**Comment by fengtan** [comment]
So, there is no mechanism to provide a cert file, but that is something that will be done when StartTLS is supported (see [roadmap](https://github.com/fengtan/ldap-explorer/blob/master/CONTRIBUTING.md#things-to-do)).

**Reply by fengtan** [reply]
@tlewisflood I have just made a pre-release [1.2.0-rc](https://github.com/fengtan/ldap-explorer/releases/tag/1.2.0-rc) with an interface to  use alternative certificates (among other TLS options):

![new_options](https://user-images.githubusercontent.com/1483301/202880412-eb35934f-4bc1-44a4-858a-f79ae3ac25e8.png)

Do you mind giving it a try and letting me know if that works for you ? You can download the `.vsix` asset from the pre-release page above and install it locally like so:

![install_vsix](https://user-images.githubusercontent.com/1483301/202880388-7a264240-76a3-49f7-9552-145573437a67.png)

**Reply by tlewisflood** [reply]
Worked like a charm! However, while it was obvious from your instructions, it can't be overstated that for windows users, the user needs to export the cert(s) (exercise for the user to figure out which) from windows so you can refer to them in the extension by full path name. So I had to add a few cert entries (since I didn't feel like guessing which one was the "correct" one), like this c:\Users\username\certs\blah1.cer, etc.

**Reply by fengtan** [reply]
Thanks for testing !

Would supporting globs (e.g. you'd be able to refer to certs as `c:\Users\username\certs\*.cer` in the extension) make it easier ?

Also can you confirm that configuring trusted certs globally (rather than for each connection) makes sense ?

**Comment by fengtan** [comment, marked as answer]
v1.2.0 has been released and includes the new TLS feaures https://github.com/fengtan/ldap-explorer/releases/tag/1.2.0

Documentation has been added to `README.md` https://github.com/fengtan/ldap-explorer#tls
